source/fundamentals/builders/sort.txt

```diff
@@ -11,3 +11,332 @@ Sorts Builders
    :class: singlecol
 
 .. _sorts-builders:
+
+Overview
+--------
+
+In this guide, we show you how to specify **sort criteria** for your
+queries using **builders**.
+
+Sort criteria are the rules MongoDB uses to sort your data. Some
+examples of sort criteria are:
+
+* Smallest number to largest number
+* Earliest time of day to latest time of day
+* Alphabetical order by first name 
+
+The builder pattern allows you to create complex immutable objects
+separately from their representation. :doc:`Builders </fundamentals/builders/>` 
```

Comment:
The use of the term `builder` here is in a different sense that the builder pattern described in Design Patterns book by the Gang of Four.  See https://en.wikipedia.org/wiki/Builder_pattern#:~:text=The%20builder%20pattern%20is%20a,Gang%20of%20Four%20design%20patterns.

So we should not refer to the builder pattern when describing any of the driver's builder classes.

Reply:
Thanks for the note. I will not reference the GoF builder pattern when discussing builder classes going forward. Thank you for clarifying. Going forward, would you say a reasonable definition of "builders" as used in the MongoDB Java driver is:

>  Builders are MongoDB Java driver classes that provide static utility methods for constructing complex `Bson` objects. 

Fix: https://github.com/mongodb/docs-java/pull/59/commits/d4c3b949928646e5c7d6000e32cb0856b7fb44d2

Reply:
That's a good question.  I think calling them `complex` is overstating the case, but I'm struggling to come up with a good definition.  Maybe if I explain in less succinct terms why we created them, you can come up with something good? We created the builder classes to make it easier for Java developers to use the CRUD API without having to refer to MongoDB reference documentation.  Instead, they can just look at a Java API that is already loaded into their IDE.  This has a few benefits:

1. It leverages the power of the IDE to do discovery and code completion
2. It leverages the power of the Java compiler and the IDE to find errors during development.  If you misspell a method name the IDE will nag you and the compiler will yell at you.  Contrast `Filters.ltd("foo")` with `new Document("$ltd", "foo")`.  In the former Intellij will underline `ltd` and give you a red bar on the right.  And you are not likely to have made the mistake in the first place because of code completion.  In the latter you get no visual indication of a problem and will only find it if you run the code.

It doesn't find all errors: in both you could misspell the field name `"foo"`.  But it helps.

Reply:
Hi Jeff, thank you providing this helpful background information. [I have filed a JIRA ticket](https://jira.mongodb.org/browse/DOCSP-15177) to better define builders, and to unify our approach to linking to builder definition in our builders subsections. I have removed `complex` for the sorts page, and have directed traffic to the main Fundamentals > Builders page which will be updated in the newly filed JIRA ticket to include this information about Code Completion/Discovery and Error Checking.

Fix: https://github.com/mongodb/docs-java/pull/59/commits/91154a2104485616950bda5f793e673f26f79c86